torch/csrc/jit/passes/onnx/shape_type_inference.cpp

```diff
@@ -1430,10 +1433,27 @@ void ONNXShapeTypeInference(
 
 } // namespace
 
+// For some operators, there are some inputs not related to shape inference.
+// For example, LSTM input 4 (sequence_lens) is optional,
+// and the shape inference can be done through other required inputs.
+// When we compute reliable, we don't need this input be reliable.
+static std::unordered_map<std::string, std::unordered_set<int64_t>> non_required_shape_inference_idx_map = {
+    {"onnx::LSTM", {4}}};
```

Comment:
Is it possible to identify all optional inputs and ignore them all, instead of specifying one index, like 4 here?

I'm concerned this solution is a little tricky because the inputs of LSTM might be changed sometime in future.

Reply:
Optional inputs are defined in onnx spec, so we have to manually identify all optional inputs, which is very costly. I would suggest we do this in a needed-basis for now. If there is further changes (which should be very rare), we will redesign this part.